src/main/java/functions/AnalyseCommand.java

```diff
@@ -0,0 +1,84 @@
+package functions;
+
+public class AnalyseCommand {
+
+    protected String commandType;
+    protected String commandDescription;
+    protected String commandTime;
+
+    public AnalyseCommand(String input) {
+        setCommandType(input);
+        setCommandDescription(input);
+        setCommandTime(input);
+    }
+
+    public void setCommandType(String input) {
+        if (input.contains(" ")) {
+            String[] commandType = input.split(" ");
+            this.commandType = commandType[0].toLowerCase();
+        } else {
+            this.commandType = input.toLowerCase();
+        }
+    }
+
+    public String getCommandType(){
+        return this.commandType;
+    }
+
+    public void setCommandDescription(String input) {
+        if (input.contains(" ")) {
+            String[] commandDescription = input.split(" ", 2);
+            if (isEvent()||isDeadline()) {
+                String[] descriptionWithoutTime = commandDescription[1].split("/");
+                this.commandDescription = descriptionWithoutTime[0];
+            } else {
+                this.commandDescription = commandDescription[1];
```

Comment:
Slightly unreadable as it is hard to tell what you mean by this. Perhaps refactoring out into methods would be a good idea.